src/libponyc/ast/parser.c

```diff
@@ -1052,11 +1052,8 @@ DEF(field);
   TOKEN("field name", TK_ID);
   SKIP("mandatory type declaration on field", TK_COLON);
   RULE("field type", type);
-  IF(TK_DELEGATE, RULE("delegated type", provides));
   IF(TK_ASSIGN, RULE("field value", infix));
-  // Order should be:
-  // id type value delegate_type
-  REORDER(0, 1, 3, 2);
+  AST_NODE(TK_NONE); // provides
```

Comment:
I think removing the blank provides node added to fields by the object litteral sugar will fix the problem. It's in `pass/sugar.c` line 840.